homeassistant/components/decora_wifi/light.py

```diff
@@ -2,127 +2,45 @@
 from __future__ import annotations
 
 import logging
+from typing import Any
 
-# pylint: disable=import-error
-from decora_wifi import DecoraWiFiSession
-from decora_wifi.models.person import Person
-from decora_wifi.models.residence import Residence
-from decora_wifi.models.residential_account import ResidentialAccount
-import voluptuous as vol
-
-from homeassistant.components import persistent_notification
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     ATTR_TRANSITION,
-    PLATFORM_SCHEMA,
     SUPPORT_BRIGHTNESS,
     SUPPORT_TRANSITION,
     LightEntity,
 )
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-_LOGGER = logging.getLogger(__name__)
-
-# Validation of the user's configuration
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Unless something has changed in the rules, when adding `config_flow` the old YAML-based config must be left in, marked deprecated, and imported into the new `config_flow`. That code will stick around for ~3 months to give users the ability to transition to the new config.

I recently did one of these conversions. Here is a link to the code https://github.com/home-assistant/core/tree/9dea96399f36f1eef2e3aafc5cf86549e437bab9/homeassistant/components/environment_canada. Take a look at the code in `__init__.py` for how to handle importing the YAML.

The link is from late last year as the code is removed now (more than 3 months passed)